src/components/protocol_handler/src/handshake_handler.cc

```diff
@@ -242,10 +256,28 @@ void HandshakeHandler::ProcessFailedHandshake(BsonObject& params,
   } else {
     service_status_update_handler_.OnServiceUpdate(
         this->connection_key(), context_.service_type_, service_status);
+
+    auto gen_reason_msg =
+        [](const ServiceStatus service_status) -> std::string {
+      switch (service_status) {
+        case ServiceStatus::PTU_FAILED:
+          return "Policy Table Update failed";
+        case ServiceStatus::CERT_INVALID:
+          return "Invalid certificate";
+        case ServiceStatus::INVALID_TIME:
+          return "Failed to get system time";
+        case ServiceStatus::UNSECURE_START_FAILED:
+          return "Unsecure start failed";
+        default:
+          return "Unknown cause of failure";
```

Comment:
I don't see any references to `ProcessFailedHandshake` with any service status besides the top three (not even UNSECURE_START_FAILED)

Reply:
Yes, its for future proofing. I think even if we add a call for `ProcessFailedHandshake` for `UNSECURED_START_FAILED` in the future, updating the reason_msg is change which might be easily missed. Let me know what you think.

Reply:
Although it looks like this case would not be handled in the `hanshake_handler` so I've removed `UNSECURE_START_FAILED` in https://github.com/smartdevicelink/sdl_core/pull/3462/commits/c289ecf066bafed2451d064f0ed4e1799b968705